From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Steve Ellcey <sellcey@caviumnetworks.com>
Date: Tue, 22 Aug 2017 16:22:05 -0700
Subject: [PATCH] aarch64: Use PTR_REG macro to fix ILP32 bug and make code
 consistent

        * sysdeps/aarch64/dl-tlsdesc.S (_dl_tlsdesc_dynamic):
        Use PTR_REG macro in cmp instruction.

(cherry picked from commit 5a706f649de3952271930a8340db4ca8aa50f485)
Signed-off-by: Anchal Agarwal <anchalag@amazon.com>

diff --git a/sysdeps/aarch64/dl-tlsdesc.S b/sysdeps/aarch64/dl-tlsdesc.S
index 67f50bd0d1fc79964558359c01d89d52e8476ca3..1d2008cbf2d339908b9477351b722e64f8a21eee 100644
--- a/sysdeps/aarch64/dl-tlsdesc.S
+++ b/sysdeps/aarch64/dl-tlsdesc.S
@@ -165,7 +165,7 @@ _dl_tlsdesc_dynamic:
 	ldp	PTR_REG (2), PTR_REG (3), [x1,#TLSDESC_MODID]
 	add	PTR_REG (0), PTR_REG (0), PTR_REG (2), lsl #(PTR_LOG_SIZE + 1)
 	ldr	PTR_REG (0), [x0] /* Load val member of DTV entry.  */
-	cmp	x0, #TLS_DTV_UNALLOCATED
+	cmp	PTR_REG (0), #TLS_DTV_UNALLOCATED
 	b.eq	2f
 	sub	PTR_REG (3), PTR_REG (3), PTR_REG (4)
 	add	PTR_REG (0), PTR_REG (0), PTR_REG (3)
